src/fields/core/fieldSubmit.vue

```diff
@@ -15,10 +15,13 @@ export default {
 				// prevent a <form /> from having it's submit event triggered
 				// when we have to validate data first
 				$event.preventDefault();
-				let errors = this.$parent.validate();
+				let validateAsync = objGet(this.formOptions, "validateAsync", false);
+				let errors = this.vfg.validate();
 				let handleErrors = errors => {
-					if (!isEmpty(errors) && isFunction(this.schema.onValidationError)) {
-						this.schema.onValidationError(this.model, this.schema, errors, $event);
+					if ((validateAsync && !isEmpty(errors)) || (!validateAsync && !errors)) {
```

Comment:
Why the addition of this extra conditional? Haven’t had my coffee yet, but it seems to be redundant?

Reply:
Because at this point, we still don't know if this form is valid or not.

If you mean there is already a check here
`if (errors && isFunction(errors.then))`

Because of the `isFunction(errors.then)` condition, a non-async valid result still go to the `else` clause and calls `handleErrors` function